src/webviews/apps/home/components/home-app.ts

```diff
@@ -0,0 +1,105 @@
+import { css, html, LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { when } from 'lit/directives/when.js';
+import type { OnboardingState } from '../../../home/protocol';
+import type { OnboardingItemConfiguration } from '../model/gitlens-onboarding';
+import { onboardingConfiguration } from '../model/gitlens-onboarding';
+
+@customElement('gl-home-app')
+export class GlHomeApp extends LitElement {
+	@property({ type: Object })
+	state?: OnboardingState;
+
+	static override readonly styles = css`
+		gl-progress-tracker {
+			padding: 0 20px;
+			--progress-color: var(--vscode-charts-blue);
+			--progress-background-color: var(--progress-bar-color);
+		}
+
+		gl-onboarding-item {
+			margin: 2px 0;
+			--disabled-color: var(--vscode-disabledForeground);
+			--checked-color: var(--sl-color-success-500, green);
+			--action-button-color: var(--sl-color-neutral-700, gray);
+		}
+
+		gl-onboarding-item-group {
+			--group-indent: 20px;
+		}
+		gl-onboarding-item-group.top-onboarding-group {
+			padding: 0 20px;
+			--group-indent: 0px;
+			margin-top: 6px;
+		}
+	`;
+
+	private get progress() {
+		const onboardingState = this.state;
+		if (!onboardingState) return 0;
+		return Number(
+			onboardingConfiguration.reduce(
+				(a, b) =>
+					a + Number(b.children ? this.calcStateFromChildren(b) : onboardingState[`${b.itemId}Checked`]),
+				0,
+			),
+		);
+	}
+
+	private get stepCount() {
+		return onboardingConfiguration.length;
+	}
+
+	private renderOnboardingItem(onboardingItem: OnboardingItemConfiguration) {
+		return html`
+			<gl-onboarding-item
+				.checked=${Boolean(this.state?.[`${onboardingItem.itemId}Checked`])}
+				playHref=${ifDefined(onboardingItem.playHref)}
+				infoHref=${ifDefined(onboardingItem.infoHref)}
+				infoTitle=${ifDefined(onboardingItem.infoTooltip)}
+				>${onboardingItem.title}
+			</gl-onboarding-item>
+		`;
+	}
+
+	private calcStateFromChildren(onboardingItem: OnboardingItemConfiguration) {
+		return (
+			onboardingItem.children?.reduce((a, b) => a && Boolean(this.state?.[`${b.itemId}Checked`]), true) ?? false
+		);
+	}
+
+	private renderOnboardingGroup(onboardingItem: OnboardingItemConfiguration) {
+		return html`
+			<gl-onboarding-item-group>
+				<gl-onboarding-item
+					.checked=${this.calcStateFromChildren(onboardingItem)}
+					slot=${'top'}
+					playHref=${ifDefined(onboardingItem.playHref)}
+					infoHref=${ifDefined(onboardingItem.infoHref)}
+					infoTitle=${ifDefined(onboardingItem.infoTooltip)}
+					>${onboardingItem.title}
+				</gl-onboarding-item>
+				${repeat(onboardingItem.children!, item => item.itemId, this.renderOnboardingItem.bind(this))}
+			</gl-onboarding-item-group>
+		`;
+	}
+
+	protected override render() {
+		return html` <gl-progress-tracker progress=${this.progress} stepCount=${this.stepCount}></gl-progress-tracker>
```

Comment:
This is missing the label+count with a collapsable toggle

![Screenshot 2024-07-30 at 3 03 17 PM](https://github.com/user-attachments/assets/62e90d53-f615-4f9d-9572-279e2911747b)